### PR TITLE
Update verify-blob to support DSSEs

### PR DESCRIPTION
**Pull request by rgreinho**
Adds the ability to verify DSSEs from the `verify-blob` command.

Fixes sigstore/cosign#1321

### CODE REVIEW

**Comment by dlorenc** [comment]
cc @priyawadhwa

**Comment by rgreinho** [comment]
Using the signature and the envelope I put in the [issue](https://github.com/sigstore/cosign/issues/1321#issuecomment-1018678455), I used it like that:

```
$ ./cosign verify-blob --key cosign.pub --signature dsse.sig dsse.sig
Verified OK
```

File: cmd/cosign/cli/verify/verify_blob.go [hunk 1]
```diff
@@ -162,9 +166,28 @@ func VerifyBlobCmd(ctx context.Context, ko sign.KeyOpts, certRef, certEmail, cer
 		}
 	}
 
-	// verify the signature
-	if err := verifier.VerifySignature(bytes.NewReader([]byte(sig)), bytes.NewReader(blobBytes)); err != nil {
-		return err
+	// Is DSSE?
+	DSSEenvelope := ssldsse.Envelope{}
+	if err := json.Unmarshal(blobBytes, &DSSEenvelope); err == nil {
```
Comment (priyawadhwa) [inline cmd/cosign/cli/verify/verify_blob.go:171]: Could we move this into a separate function and add a unit test for it?

**Review by priyawadhwa — COMMENTED** [review]
Thanks for adding this @rgreinho ! Just added one comment about unit testing :)

File: cmd/cosign/cli/verify/verify_blob.go (hunk 1 above)
Comment (rgreinho) [reply cmd/cosign/cli/verify/verify_blob.go]: Ok, I found some help in the sigstore package, no the code is shorter and cleaner.

File: cmd/cosign/cli/verify/verify_blob.go [hunk 2]
```diff
@@ -162,6 +167,11 @@ func VerifyBlobCmd(ctx context.Context, ko sign.KeyOpts, certRef, certEmail, cer
 		}
 	}
 
+	// Use the DSSE verifier if the payload is a DSSE with the In-Toto format.
+	if isIntotoDSSE, _ := isIntotoDSSE(blobBytes); isIntotoDSSE {
```
Comment (priyawadhwa) [inline cmd/cosign/cli/verify/verify_blob.go:171]: let's either check the error here and return it, or change this function to only return a `bool` if the error should be ignored :)

File: cmd/cosign/cli/verify/verify_blob_test.go [hunk 3]
```diff
@@ -93,3 +95,45 @@ func TestSignaturesBundle(t *testing.T) {
...
+	}
+}
+
+func TestIsIntotoDSSEWithBytes(t *testing.T) {
+	tts := []struct {
+		envelope     []byte
+		isIntotoDSSE bool
+	}{
+		{
+			envelope:     []byte("This is no valid"),
```
Comment (priyawadhwa) [inline cmd/cosign/cli/verify/verify_blob_test.go:128]: could you add a test with a regular signature in here just to make sure it still fails?

File: cmd/cosign/cli/verify/verify_blob.go (hunk 2 above)
Comment (rgreinho) [reply cmd/cosign/cli/verify/verify_blob.go:171]: Copy that! Addressed ✅

File: cmd/cosign/cli/verify/verify_blob_test.go (hunk 3 above)
Comment (rgreinho) [reply cmd/cosign/cli/verify/verify_blob_test.go:128]: Oki doki. I think I did what you wanted, my apologies if I misunderstood.

**Review by priyawadhwa — APPROVED** [review]
LGTM, thanks @rgreinho !

**Pull request merged**